### PR TITLE
Add an API to exchange refresh -> access token

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 1]
```diff
@@ -19,6 +19,26 @@ service SecurityTokenService {
...
+        get: "/sts/access_token"
+      }
+    };
+  }
+}
+
+message GetAccessTokenRequest {
+  repeated string aud = 1;
+  string scope = 2;
+  string refreshToken = 3;
```
Comment (imjasonh) [inline proto/platform/oidc/v1/oidc.platform.proto:36]: I thought proto style was `refresh_token`?

Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto]: Thanks! Updated.

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 2]
```diff
@@ -19,6 +19,26 @@ service SecurityTokenService {
...
+}
+
+message GetAccessTokenRequest {
+  repeated string aud = 1;
+  string scope = 2;
+  string refresh_token = 3;
+
+  // List of capabilities to request for the token.
+  repeated string cap = 4 [(google.api.resource_reference) = {
+    type: "chainguard.capabilities/Capability"
```
Comment (mattmoor) [inline proto/platform/oidc/v1/oidc.platform.proto:40]: Hmm, I'm not familiar with this syntax.  Why not a repeated enum field?  Did we change something about capabilities?

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 3]
```diff
@@ -19,6 +19,26 @@ service SecurityTokenService {
       }
     };
   }
+
+  rpc GetAccessToken(GetAccessTokenRequest) returns (TokenPair) {
+    option (google.api.http) = {
+      post: "/sts/access_token"
+      additional_bindings {
+        get: "/sts/access_token"
+      }
```
Comment (mattmoor) [inline proto/platform/oidc/v1/oidc.platform.proto:28]: Why?

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 4]
```diff
@@ -19,6 +19,26 @@ service SecurityTokenService {
...
+    };
+  }
+}
+
+message GetAccessTokenRequest {
+  repeated string aud = 1;
+  string scope = 2;
+  string refresh_token = 3;
```
Comment (mattmoor) [inline proto/platform/oidc/v1/oidc.platform.proto:36]: Should this be in the `Authorization` header?

File: proto/platform/oidc/v1/oidc.platform.proto (hunk 2 above)
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto:40]: This is copy/pasta from current `ExchangeRequest`

File: proto/platform/oidc/v1/oidc.platform.proto (hunk 4 above)
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto:36]: The current `ExchangeRequest` does it through a field

File: proto/platform/oidc/v1/oidc.platform.proto (hunk 3 above)
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto]: Removed

File: proto/platform/oidc/v1/oidc.platform.proto (hunk 4 above)
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto]: Fixed

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 5]
```diff
@@ -19,6 +19,22 @@ service SecurityTokenService {
...
   }
+
+  rpc GetAccessToken(GetAccessTokenRequest) returns (TokenPair) {
+    option (google.api.http) = {
+      post: "/sts/access_token"
+    };
+  }
+}
+
+message GetAccessTokenRequest {
```
Comment (tcnghia) [inline proto/platform/oidc/v1/oidc.platform.proto:30]: Could also use `Refresh` here as potentially a better name

Comment (mattmoor) [reply proto/platform/oidc/v1/oidc.platform.proto:30]: For consistency maybe we should use `ExchangeRefreshToken`

**Review by mattmoor — APPROVED** [review]

File: proto/platform/oidc/v1/oidc.platform.proto (hunk 5 above)
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto]: Done!

File: proto/platform/oidc/v1/oidc.platform.proto [hunk 6]
```diff
@@ -39,3 +55,10 @@ message ExchangeRequest {
 message RawToken {
   string token = 1;
 }
+
+// GetAccessToken returns a pair of token, in order to allow
+// refresh token to also be rotated.
+message TokenPair {
+  string token = 1;
```
[the remark this answers is not in the document]
Comment (tcnghia) [reply proto/platform/oidc/v1/oidc.platform.proto]: Done!

**Pull request merged**